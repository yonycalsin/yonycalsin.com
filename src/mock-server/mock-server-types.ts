import type { DefaultBodyType, PathParams, ResponseComposition, RestContext, RestRequest } from 'msw'

export type MockRequest<Body = never, Params extends PathParams = PathParams<string>> = RestRequest<Body, Params>

export type MockResponse = ResponseComposition<DefaultBodyType>

export type MockContext = RestContext
