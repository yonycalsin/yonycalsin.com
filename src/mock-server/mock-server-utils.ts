function sanitizeEndpoint(endpoint: string) {
  const url = new URL(endpoint)

  if (!url.search) {
    return url.href
  }

  return url.href.replace(url.search, '')
}

export { sanitizeEndpoint }
