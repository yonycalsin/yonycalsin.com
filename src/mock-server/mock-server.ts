export async function initMocks() {
  if (typeof window === 'undefined') {
    const { server } = await import('./mock-server-node')

    server.listen()
  } else {
    const { worker } = await import('./mock-server-browser')

    worker.start()
  }
}
