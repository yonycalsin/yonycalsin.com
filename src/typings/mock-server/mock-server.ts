import type { DefaultBodyType, PathParams, ResponseComposition, RestContext, RestRequest } from 'msw'

type MockRequest<Body extends DefaultBodyType = never, Params extends PathParams = PathParams<string>> = RestRequest<
  Body,
  Params
>

type MockResponse = ResponseComposition<DefaultBodyType>

type MockContext = RestContext

export type { MockContext, MockRequest, MockResponse }
