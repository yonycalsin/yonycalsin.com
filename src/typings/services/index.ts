export * from './blog'
export * from './shared'
