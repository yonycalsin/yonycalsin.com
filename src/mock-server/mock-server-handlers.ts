import { rest } from 'msw'

import { API_ENDPOINTS } from 'services/shared'
import {
  getAllAchievementsApi,
  getAllBooksApi,
  getAllPagesApi,
  getAllProjectsApi,
  getCategoriesApi,
  getCategoryApi,
  getFeaturedAchievementsApi,
  getFeaturedRecommendationsApi,
  getPageApi,
  getPinnedProjectsApi,
  getPostApi,
  getPostsApi,
  getReadingBooksApi,
} from './mocks'

export const mockServerHandlers = [
  // Achievements
  rest.get(API_ENDPOINTS.ALL_ACHIEVEMENTS, getAllAchievementsApi),
  rest.get(API_ENDPOINTS.FEATURED_ACHIEVEMENTS, getFeaturedAchievementsApi),

  // Posts
  rest.get(API_ENDPOINTS.POST(':slug'), getPostApi),
  rest.get(API_ENDPOINTS.POSTS, getPostsApi),

  // Categories
  rest.get(API_ENDPOINTS.CATEGORY(':slug'), getCategoryApi),
  rest.get(API_ENDPOINTS.CATEGORIES, getCategoriesApi),

  // Books
  rest.get(API_ENDPOINTS.ALL_BOOKS, getAllBooksApi),
  rest.get(API_ENDPOINTS.READING_BOOKS, getReadingBooksApi),

  // Pages
  rest.get(API_ENDPOINTS.PAGE(':slug'), getPageApi),
  rest.get(API_ENDPOINTS.ALL_PAGES, getAllPagesApi),

  // Projects
  rest.get(API_ENDPOINTS.ALL_PROJECTS, getAllProjectsApi),
  rest.get(API_ENDPOINTS.PINNED_PROJECTS, getPinnedProjectsApi),

  // Recommendations
  rest.get(API_ENDPOINTS.FEATURED_RECOMMENDATIONS, getFeaturedRecommendationsApi),
]
