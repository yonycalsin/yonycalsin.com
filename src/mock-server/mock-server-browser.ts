import { setupWorker } from 'msw'

import { mockServerHandlers } from './mock-server-handlers'

export const worker = setupWorker(...mockServerHandlers)
