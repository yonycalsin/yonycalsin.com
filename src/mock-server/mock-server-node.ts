import { setupServer } from 'msw/node'

import { mockServerHandlers } from './mock-server-handlers'

function initServer() {
  const server = setupServer(...mockServerHandlers)

  return server
}

export default initServer
