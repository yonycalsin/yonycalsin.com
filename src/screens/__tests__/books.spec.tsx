import { screen, within } from '@testing-library/react'

import { allBooksSuccess, readingBooksSuccess } from 'mock-server/mocks'
import { setupWithReactQuery, TestProvider } from 'tests'
import { getAllBooksApi, getReadingBooksApi } from 'services'
import BooksScreen from 'screens/books'

function setup() {
  return setupWithReactQuery(
    <TestProvider>
      <BooksScreen />
    </TestProvider>,
  )
}

jest.mock('services/books')

const mockGetAllBooksApi = getAllBooksApi as jest.Mock

const mockGetReadingBooksApi = getReadingBooksApi as jest.Mock

describe('BooksScreen', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders the books screen', async () => {
    mockGetAllBooksApi.mockReturnValueOnce(allBooksSuccess)

    mockGetReadingBooksApi.mockReturnValueOnce(readingBooksSuccess)

    const view = setup()

    await screen.findByRole('progressbar')

    const allBooksList = await screen.findByRole('list', { name: /list of all books/i })

    const allBooksListItems = within(allBooksList).getAllByRole('listitem')

    expect(allBooksListItems).toHaveLength(allBooksSuccess.data.length)

    const allReadingBooksList = await screen.findByRole('list', { name: /list of reading books/i })

    const allReadingBooksListItems = within(allReadingBooksList).getAllByRole('listitem')

    expect(allReadingBooksListItems).toHaveLength(readingBooksSuccess.data.length)

    expect(view.baseElement).toMatchSnapshot()
  })
})
