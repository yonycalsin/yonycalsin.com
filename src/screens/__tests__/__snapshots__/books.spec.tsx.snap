// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksScreen renders the books screen 1`] = `
<body>
  <div>
    <div
      class="w-full z-10 shadow-sm"
    >
      <div
        class="container py-5 md:py-6 flex justify-between items-center"
      >
        <a
          href="/"
        >
          <img
            alt="Logo"
            data-nimg="1"
            decoding="async"
            height="32"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=64&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=64&q=75 2x"
            style="color: transparent;"
            width="32"
          />
        </a>
        <div
          class="space-x-4"
        >
          <a
            class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
            href="/projects"
          >
            Projects
          </a>
          <a
            class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
            href="/me"
          >
            About Me
          </a>
        </div>
      </div>
    </div>
    <main
      class="container space-y-6 py-10"
    >
      <header
        class="space-y-3"
      >
        <h1
          class="font-bold font-heading text-4xl"
        >
          Books (
          1
          )
        </h1>
        <p>
          This page contains the books I have read along of my life.
        </p>
      </header>
      <div
        class="space-y-3"
      >
        <h1
          class="font-bold font-heading text-2xl"
        >
          Reading:
        </h1>
        <div
          aria-label="List of reading books"
          class="grid grid-cols-2 lg:grid-cols-4 gap-6"
          role="list"
        >
          <div
            class="space-y-3 p-3 bg-white rounded-lg shadow-card"
            role="listitem"
          >
            <div>
              <img
                alt="Clean Architecture"
                class="w-full rounded-md"
                src="https://static.yonycalsin.com/books/clean-arquitecture-front.png"
              />
            </div>
            <h1
              class="font-bold font-heading text-base"
            >
              Clean Architecture
            </h1>
            <p
              class="text-sm line-clamp-1"
            >
              Mark Zuck
            </p>
            <div
              class="flex flex-row items-center space-x-2"
            >
              <svg
                class="text-primary-600"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                />
              </svg>
              <span>
                Rating 
                4.5
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="space-y-3"
      >
        <h1
          class="font-bold font-heading text-2xl"
        >
          All:
        </h1>
        <div
          aria-label="List of all books"
          class="grid grid-cols-2 lg:grid-cols-4 gap-6"
          role="list"
        >
          <div
            class="space-y-3 p-3 bg-white rounded-lg shadow-card"
            role="listitem"
          >
            <div>
              <img
                alt="Clean Code"
                class="w-full rounded-md"
                src="https://static.yonycalsin.com/books/clean-code-front.png"
              />
            </div>
            <h1
              class="font-bold font-heading text-base"
            >
              Clean Code
            </h1>
            <p
              class="text-sm line-clamp-1"
            >
              Tim Chuckby
            </p>
            <div
              class="flex flex-row items-center space-x-2"
            >
              <svg
                class="text-primary-600"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 1024 1024"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M908.1 353.1l-253.9-36.9L540.7 86.1c-3.1-6.3-8.2-11.4-14.5-14.5-15.8-7.8-35-1.3-42.9 14.5L369.8 316.2l-253.9 36.9c-7 1-13.4 4.3-18.3 9.3a32.05 32.05 0 0 0 .6 45.3l183.7 179.1-43.4 252.9a31.95 31.95 0 0 0 46.4 33.7L512 754l227.1 119.4c6.2 3.3 13.4 4.4 20.3 3.2 17.4-3 29.1-19.5 26.1-36.9l-43.4-252.9 183.7-179.1c5-4.9 8.3-11.3 9.3-18.3 2.7-17.5-9.5-33.7-27-36.3z"
                />
              </svg>
              <span>
                Rating 
                5
              </span>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="my-6"
    >
      <div
        class="container"
      >
        <footer
          class="p-4 bg-white rounded-lg shadow flex flex-col md:flex-row md:items-center md:justify-between md:p-6"
        >
          <span
            class="text-gray-500"
          >
            © 2022
             
            <a
              class="hover:underline"
              href="https://yonycalsin.com"
            >
              Yony Calsin™
            </a>
            . All Rights Reserved.
          </span>
          <ul
            class="flex flex-wrap gap-3 items-center justify-center"
          >
            <li>
              <a
                class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
                href="/me"
              >
                About
              </a>
            </li>
            <li>
              <a
                class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
                href="/contact"
              >
                Contact
              </a>
            </li>
          </ul>
        </footer>
      </div>
    </footer>
    <div
      class="space-y-5 fixed bottom-8 left-8 hidden lg:flex flex-col"
    >
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://github.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Github
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://www.linkedin.com/in/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Linkedin
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="https://twitter.com/yonycalsin"
        rel="noreferrer"
        target="_blank"
      >
        Twitter
      </a>
      <a
        class="font-medium text-primary-600 dark:text-primary-500 hover:underline"
        href="mailto:helloyonycalsin@gmail.com"
        rel="noreferrer"
        target="_blank"
      >
        Email
      </a>
    </div>
  </div>
</body>
`;
